`r if(knitr:::is_latex_output()) '\\appendix'`

`r if(!knitr:::is_latex_output()) '# (APPENDIX) Appendix {-}'` 

# Transfer of regulatory knowledge from human to mouse for functional genomics analysis
## Supplementary Figures
```{r transfer-sfig-1, fig.cap = "Results of pathway-wise ROC-curves analysis. The dashed line indicate the performance of a random model. Missing mouse or human ROC-curves are due to missing benchmark data for the corresponding pathway."}
knitr::include_graphics(here::here("data/conserved_footprints/FigureS1.png"))
```

```{r transfer-sfig-2, fig.cap = "(A) Barplot showing the imbalance of true negatives (TN) and true positives (TP) in our benchmark dataset for human and mouse. (B) Distribution of AUROC’s computed for human and mouse separately from a balanced dataset (generated by downsampling the TN to equal the number of TP). The diamonds indicate the AUROC of the unbalanced dataset."}
knitr::include_graphics(here::here("data/conserved_footprints/FigureS2.png"))
```

```{r transfer-sfig-3, fig.cap = "Results of pathway-wise PR-curves analysis. The dashed line indicates the performance of a random model. Missing mouse or human PR-curves are due to missing benchmark data for the corresponding pathway."}
knitr::include_graphics(here::here("data/conserved_footprints/FigureS3.png"))
```

```{r transfer-sfig-4, fig.cap = "(A) Mouse-DoRothEA properties showing number of transcription factors (TF), targets, and interactions itemized by confidence level. (B) Overlap of TFs between mouse-DoRothEA and TRRUST. (C) Similarity analysis of target genes for each overlapping TF between mouse-DoRothEA and TRRUST. Jaccard index and overlap coefficient were used to quantify similarity. Color and size indicate if and how often the TF was covered in the benchmark data. (D) Number of TFs covered in the benchmark dataset by intersection of mouse-DoRothEA and TRRUST dependent of the TF-confidence level."}
knitr::include_graphics(here::here("data/conserved_footprints/FigureS4.png"))
```

```{r transfer-sfig-5, fig.cap = "(A) Barplot showing the imbalance of true negatives (TN) and true positives (TP) in our benchmark dataset for mouse-DoRothEA filtered for confidence level A or B. (B) Distribution of AUROC’s computed for DoRothEA and TRRUST separately from a balanced dataset (generated by downsampling the TN to equal the number of TP). The diamonds indicate the AUROC of the unbalanced dataset."}
knitr::include_graphics(here::here("data/conserved_footprints/FigureS5.png"))
```

# Robustness and applicability of transcription factor and pathway analysis tools on single-cell RNA-seq data 
## Supplementary Figures
```{r scbench-fig-s1, fig.cap = "a Workflow to induce low gene coverage with a subsequent benchmark of the tools PROGENy, DoRothEA and GO-GSEA against low gene coverage. b,c,e Scatterplot showing how well AUROC and AUPRC of b DoRothEA (AB), c PROGENy with 100 footprint genes per pathway, and GO-GSEA are correlated. The labels correspond to the gene coverages. d Mapping table between PROGENy pathways and GO terms/GO IDs."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure1.png"))
```

```{r scbench-fig-s2, fig.cap = "Pathway-wise evaluation of a PROGENy and b GO-GSEA at different gene coverages. Performance is measured as Area under the ROC curve (AUROC). The dashed line indicates the performance of a random model. The colors in b are meant only as a visual support to distinguish between the individual violin plots and jittered points."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure2.png"))
```

```{r scbench-fig-s3, fig.cap = "Overview of the benchmark dataset of the in silico study for a TF and b pathway analysis tools. The term coverage denotes the number of distinct perturbed TFs and pathways in the benchmark dataset covered by the respective gene set resource. As individual pathways/TFs can be perturbed several times in independent experiments we also provide the total number of perturbation experiments. In the case of TF perturbation experiments we also provide DoRothEA’s confidence class for each perturbed TF indicating the quality of its regulon within DoRothEA (A - high quality to E - low quality)."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure3.png"))
```

```{r scbench-fig-s4, fig.cap = "Comparison of single-cell-specific properties between real and simulated single cells. a Count distribution of a representative gene for a real and a simulated single cell. b Mean-variance relationship of gene expression of a representative data set for a real and a simulated single cell. c The dependence of the number of detected genes in a real and a simulated single cell on the library size."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure4.png"))
```

```{r scbench-fig-s5, fig.cap = "Variance in the performance (measured as AUROC) of a DoRothEA, b D-AUCell, c metaVIPER, d PROGENy and e P-AUCell on single cells for different combinations of simulation parameters. The variance is calculated by repeating the simulation of each single-cell for each parameter combination 25 times."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure5.png"))
```

```{r scbench-fig-s6, fig.cap = "Scatterplot comparing the performance of a DoRothEA, b D-AUCell, c metaVIPER, d PROGENy, and e P-AUCell on single cells and bulk, measured with AUROC and AUPRC with respect to different combinations of a,b,c DoRothEA’s confidence levels or d,e different number of footprint genes per pathway."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure6.png"))
```

```{r scbench-fig-s7, fig.cap = "Effect of the simulation parameters on the performance of TF and pathway analysis tools. The tile plots show the difference in performance of a DoRothEA, b D-AUCell, c metaVIPER, d PROGENy, and e P-AUCell between single cells and corresponding bulk samples, a,b,c across all confidence level combinations or d,e different number of footprint genes per pathway. A negative value indicates that the performance on bulk was better than on the simulated single cells and vice versa. The letters/numbers within the tiles indicates which confidence level combination/number of footprint genes per pathway performed the best on the single-cell data for the given parameter combination. The tile marked in red corresponds to the parameter setting used for previous plots in the main manuscript."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure7.png"))
```

```{r scbench-fig-s8, fig.cap = "a Overview of the in-vitro benchmark dataset. The term coverage denotes the number of distinct perturbed TFs in the benchmark datasets. As individual TFs can be perturbed several times in independent experiments we also provide the total number of perturbation experiments. We also provide DoRothEA’s confidence class for each TF indicating the quality of its regulon (A - high quality to E - low quality). b The dependence of the number of detected genes on the library size for all benchmark datasets. The number of corresponding cells are displayed as well. c logFC of perturbed target/TF for the corresponding perturbation experiment for all benchmark datasets. d Distribution of logFC of all genes for each benchmark dataset. e Relationship between AUROC and AUPRC for DoRothEA, D-AUCell and metaVIPER with respect to different combinations of DoRothEA’s confidence levels for each benchmark dataset."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure8.png"))
```

```{r scbench-fig-s9, fig.cap = "Overlap of TF regulon resources. a Overlapping TFs of protocol-specific SCENIC regulatory networks. All 13 networks share 24 TFs. b Overlapping TFs between protocol-specific SCENIC regulatory networks, GTEx regulons and DoRothEA. All resources share 20 TFs. The remaining vertical bar plots indicate the number of TFs that are exclusive for the respective regulon resource. The horizontal bar plots indicate the total number of TFs for the regulon resource."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure9.png"))
```

```{r scbench-fig-s10, fig.cap = "Pairwise (Pearson) correlations of TF activities between the scRNA-technologies for each TF analysis tool."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure10.png"))
```

```{r scbench-fig-s11, fig.cap = "Identification of the best method to determine the top 2000 highly variable genes to be considered for dimensionality reduction. We tested three different selection methods implemented in Seurat (disp = dispersion, mvp = mean.var.plot, vst). We also included CV (squared coefficient of variation - (sd/mean)**2) and MVG (most variable genes - genes with the highest variance). Those methods are compared to the case of considering the full gene expression matrix for dimensionality reduction, indicated here as ‘Normalized expression’."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure11.png"))
```

```{r scbench-fig-s12, fig.cap = "a Statistical analysis of cell cluster purity in the reduced space: i) differences in the quality of clustering with respect to the positive control and ii) quality of clustering of scRNA-seq protocols in contrast to Quartz-Seq2 for TF activity tools. This analysis was performed independently for all hierarchy levels (Hrchy. Lvl.). The legend key ‘estimate’ corresponds to the estimated coefficients of the linear model. A negative value indicates a worse performance than the reference level (positive control for input matrices and Quartz-Seq2 for protocols) and vice versa.The dashed line indicates a p-value of 0.05. b Comparison of cluster purity measured by the silhouette widths obtained when considering highly variable genes identified by Seurat, TF analysis tools and controls for all hierarchy levels."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure12.png"))
```

```{r scbench-fig-s13, fig.cap = "a Statistical analysis of cell cluster purity in the reduced space : i) differences in the quality of clustering using different input matrices with respect to positive control and ii) quality of clustering of scRNA-seq protocol in contrast to Quartz-Seq2 for pathway activity inference tools. This analysis was performed independently for all hierarchy levels (Hrchy. Lvl.). The legend key ‘estimate’ corresponds to the estimated coefficients of the linear model. A negative value indicates a worse performance than the reference level (positive control for input matrices and Quartz-Seq2 for protocols) and vice versa. The dashed line indicates a p-value of 0.05. b Comparison of cluster purity measured by the silhouette widths obtained when considering highly variable genes defined by Seurat, pathway analysis tools and controls for all hierarchy levels."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure13.png"))
```

```{r scbench-fig-s14, fig.cap = "Selected TF and pathway activities of various tools inferred from the Quartz-Seq2 gene expression data summarized for each cell type/cell line separately. The letters in the brackets correspond to DoRothEA’s confidence levels and the numbers in brackets correspond to the number of footprint genes per pathway."}
knitr::include_graphics(here::here("data/sc_benchmark/SFigure14.png"))
```

# Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice
## Supplementary Material and Methods {#liver-smm}
### Collecting, curating, and processing publicly available genome-wide transcriptomics data
The raw data of the publicly available genome-wide transcriptomic studies (.CEL files for microarrays and count matrices for RNA-seq) were downloaded from Gene Expression Omnibus (GEO; @edgar_2002) and ArrayExpress [@kauffmann_2009]. For the respective accession, identifier see Table \@ref(tab:liver-tab-1). Associated metadata was manually collected or accessed via the R/Bioconductor package GEOquery (version 2.56.0; @davis_2007). To further process the raw data of microarrays and RNA-seq a suite of different Bioconductor packages [@gentleman_2004] was used. 

#### Processing of microarray data
First, a probe-level model was fitted to the raw data to subsequently control the array quality based on the relative log expression values (RLE) and the normalized unscaled standard errors (NUSE) using the R/Bioconductor package oligo (version 1.52.0; @carvalho_2010). Arrays that deviated more than 0.1 from 0 for RLE and from 1 for NUSE were discarded due to expected poor quality. Subsequently, the raw data was normalized with the RMA algorithm, also implemented within the oligo package. Dependent on the studied organisms probe identifiers were mapped either to mouse/MGI or human/HGNC symbols using the R/Bioconductor package annotate (version 1.66.0) in combination with the individual annotation package of the used array type (e.g. hugene11sttranscriptcluster.db). If several probes matched the same gene symbol the expression was averaged. Genes with a constant expression across all samples were manually removed.

#### Processing of RNA-seq data
The preprocessing started by filtering out lowly expressed genes using the R/Bioconductor package edgeR (version 3.30.0; @robinson_2010) to increase the power of downstream statistical tests. Subsequently, the expression data was normalized by correcting for differences in library composition using also edgeR. Finally, the normalized data was transformed to log2-counts per million with the R/Bioconductor package limma (version 3.44.1; @ritchie_2015) 

### Differential gene expression analysis
The differential gene expression analysis was performed using the R/Bioconductor package limma (version 3.44.1; @ritchie_2015). Unless otherwise stated a gene is considered differentially expressed with a false discovery rate (FDR) $\le$ 0.05 and |logFC| $\ge$ 1. Experiments with a time course or case-control design were handled differently. The resulting gene signatures of a differential gene expression analysis is referred to as contrast.

#### Time course design
For the acute CCl~4~, Acetaminophen (APAP), and partial hepatectomy (PH) experiment, each time point is compared vs time point 0 to extract the effect of the respective intoxication or treatment. The chronic CCl~4~ experiment was handled differently since time-matched oil controls were available for month 2 and month 12, but not for month 6 (Figure \@ref(fig:liver-fig-1)A). As the oil effect was assumed to be constant across time the expression values for the oil sample at month 6 were imputed by averaging the expression of the oil samples at month 2 and month 12. To regress out the effect of the oil the treated samples were compared against their time-matched oil controls. Also, the Bile duct ligation (BDL) experiment was handled differently because there were time-matched sham surgery samples available for days 1, 3, and 7 but not for day 21 (Supplementary Figure \@ref(fig:liver-sfig-8)A). For this reason, the expression values for the sham surgery sample at day 21 were again imputed by averaging the expression of the remaining sham surgery samples. Finally, the effect of BDL was extracted by comparing BDL samples vs time-matched sham surgery samples.

#### Case-Control design
For the experiments following the classical case-control design which is true for the Lipopolysaccharide (LPS) and Tunicamycin mouse models and all human studies, treated samples/patients with CLD were compared against untreated/healthy samples/patients or samples of patients with lower disease stages.

### Clustering of time-series gene expression data
Time-series gene expression data were clustered with the software program STEM (Short Time-series Expression Miner, version 1.3.12; @ernst_2005; @ernst_2006). As input, log-fold changes were provided from preceded differential gene expression analysis, where each time point was compared against time point 0. Accordingly, within STEM the normalization strategy “No normalization/add 0” was selected. Expression profiles were clustered using the default STEM clustering method with a specified maximal change between time points of 10 to not exclude drastic changes in the gene expression program, given partially large time spans between individual time points. Up to 20 different model profiles were returned by STEM. All other STEM parameters were left as default.

### Comparing the similarity of gene sets via the Jaccard Index and Overlap Coefficient
The similarity or overlap between two gene sets was summarized either as Jaccard Index or Overlap Coefficient. Jaccard Index was used if the tested gene set had the same size, while for unbalanced gene set size the Overlap Coefficient was used. 
For the similarity analysis among the different acute mouse models or patient cohorts, the gene sets were composed of the top 500 differentially expressed genes based on limma’s moderated t-value.

### Testing direction of regulation with Gene Set Enrichment Analysis
To test whether the differentially expressed genes of an arbitrary study A are consistently regulated in an also arbitrary but independent study B Gene Set Enrichment Analysis (GSEA; @subramanian_2005) was performed. From study A the top 500 up-and downregulated genes were extracted based on the moderated t-statistic. If the upregulated genes of study A are enriched among the upregulated genes of study B GSEA returns a positive enrichment score (ES). For enriched downregulated genes among the downregulated genes of study B a negative ES. GSEA was performed with the R/Bioconductor package fgsea (version 1.14.0; @sergushichev_2016). If the enrichment was significant (FDR $\le$ 0.05) it is concluded that either up- and/or downregulated genes of study A are consistently regulated also in study B.

### Selection of the time point of the strongest altered expression profile
For the reversible acute time-course experiments (CCl~4~, APAP, and BDL) the time point with the strongest deregulated expression profile was determined by comparing the mean distance of the individual time points to their respective control time point along with the principal component 1 (PC1) axis in the principal component (PC) space. The time point with the largest mean distance was considered as the time point of strongest deregulation. For the bile duct ligation experiment this strategy was not applicable due to the irreversible damage. We selected 24 hours as this was comparable to the selected time points of the other acute experiments.

### Functional characterization of transcriptomic data with various gene set resources
To functionally characterize the mouse and human transcriptomic data sets different gene set resources were used, such as biological processes from Gene Ontology (GO [@ashburner_2000]), DoRothEA’s regulons of confidence level A, B, and C [@garciaalonso_2019; @holland_2020] and PROGENy’s top 100 pathway responsive-genes [@schubert_2018; @holland_2020]. GO gene sets were accessed with the R package msigdf (version 7.1; https://github.com/ToledoEM/msigdf), which queries itself the Molecular Signatures Database (MSigDB; @liberzon_2011). DoRothEA and PROGENy gene sets were accessed via their corresponding R/Bioconductor packages dorothea (version 1.0.1) and progeny (version 1.11.3).

As the statistical method overrepresentation analysis (ORA) was applied via the R function `stats::fisher.test()`. The number of background genes was set to 20,000 as this reflects a reasonable number of covered genes in transcriptomic studies. The minimum required gene set size was set to 10. The false discovery rate (FDR) is computed for each gene set resource individually to account for multiple hypothesis testing.

### Manual clustering and text analysis of GO-terms
Significantly overrepresented GO-terms (FDR $\le$ 0.05) were further analyzed by manual classification into more coarse-grained biological processes and text analysis to identify the most common keywords. Text analysis was performed with the R package tidytext (version 0.2.6; @silge_2016). Numbers and highly unspecific words such as “process” or “regulation” were removed. The frequencies of the most common keywords were visualized as word cloud with the R package ggwordcloud (version 0.5.0; https://lepennec.github.io/ggwordcloud/).

### Construction and ranking of exclusive chronic, exclusive acute, and common gene sets
To identify a set of genes that is exclusive for chronic or acute liver damage in mouse models the acute and chronic mouse studies were integrated. 

#### Construction of gene sets
First, the union of all differentially expressed genes was built across all the contrasts derived from the chronic or acute mouse experiment. A gene was considered differentially expressed with a FDR $\le$ 1e-4 and |logFC| $\ge$ 1. This conservative FDR threshold was opted for to include only the most reliable deregulated genes in this analysis. Indeed all chronic contrasts were considered. In the acute setup, however, the last time point of the bile duct ligation experiment (21 days after surgery) was removed as this damage is no longer considered to be acute. A consensus gene-level statistic was assigned for each gene in the unified chronic and acute gene set by computing the median of the t-statistics. Based on the sign of this consensus gene-level statistic genes were classified as up or downregulated. Genes that are listed only in the chronic or acute gene set were considered as exclusive chronic or exclusive acute genes, respectively. Accordingly, genes that were covered by both gene sets were deregulated in the liver irrespective of the nature of the damage. Therefore this gene set is referred to as the “common” gene set.

#### Ranking of gene sets
For these three different gene sets, a metric was computed per gene $i$ integrating statistics from the chronic and selected acute contrasts. Regarding the acute mouse models, the number of considered acute contrasts was expanded. Next to the time points with the strongest deregulated profile, all time points between 8 and 48 hours were integrated to take into account the dynamic process of liver injury and recovery.

Metric for exclusive chronic gene set: i) consensus chronic gene-level statistic $c_{consensus-chronic_{i\ }}$, ii) median t-statistic of selected acute contrasts $\mu_{acute_{i\ }}$, and iii) variance of selected acute contrasts $\sigma_{acute_{i\ }}^2$. 

$$rank\left(\left|c_{consensus-chronic_{i\ }}\cdot\ \frac{1}{\mu_{acute_{i\ }}\ }\ \cdot\ \sqrt{\frac{1}{\sigma_{acute_{i\ }}^2}}\right|\right)$$

This metric prioritizes genes that have a high consensus chronic gene-level statistic and at the same time are consistently not deregulated in selected acute contrasts.
 
Metric for exclusive acute gene set: i) consensus acute gene-level statistic $a_{consensus-acute_{i\ }}$, ii) median t-statistic of selected chronic contrasts $\mu_{chronic_{i\ }}$ and iii) variance of selected chronic contrasts $\sigma_{chronic_{i\ }}^2$. 

$$rank\left(\left|a_{consensus-acute_{i\ }}\cdot\ \frac{1}{\mu_{chronic_{i\ }}\ }\ \cdot\ \sqrt{\frac{1}{\sigma_{chronic_{i\ }}^2}}\right|\right)$$
Similar to above this metric ranks genes high that have a high consensus acute gene-level statistic and at the same time are consistently not deregulated in the chronic contrasts.

Metric for common gene set: same variables as for the exclusive chronic gene set.

$$rank\left(c_{consensus-chronic_{i\ }}\cdot\ \frac{1}{\mu_{acute_{i\ }}\ }\ \cdot\ \sqrt{\frac{1}{\sigma_{acute_{i\ }}^2}}\right)$$

This metric will rank genes high that have a high consensus chronic gene-level statistic and simultaneously are consistently and strongly regulated in the same direction as in the chronic scenario in selected acute contrasts. 

### Identification of consistently deregulated genes in patients and chronic CCl~4~ mouse model
To identify consistently deregulated genes in patients of CLD and chronic CCl~4~ mouse model the top 500 up and top 500 downregulated genes selected by the absolute value of the moderated t-statistic were extracted from each human contrast. Subsequently, those gene sets were enriched in the three different contrasts from the chronic CCl~4~ mouse model corresponding to the three different time points: 2, 6, and 12 months. For the enrichment, the R/Bioconductor package fgsea (version 1.14.0; @sergushichev_2016) was used with 1000 permutations. Afterward, the leading-edge genes were extracted from each enrichment if it met two criteria: i) FDR $\le$ 0.05 and ii) consistent regulation between human and mouse data, i.e. when the upregulated human genes were enriched in the upregulated mouse genes indicated by a positive enrichment score (ES)) and vice versa, indicated by a negative ES.
As some human studies have multiple contrasts the leading-edge genes were unified per study and chronic time point. Subsequently, those leading-edge genes were filtered for appearing per chronic time point in at least three studies.

### Mapping of consistently deregulated genes in human and mouse to cell types using single-cell RNA-sequencing data
To identify whether the consistently deregulated genes in mice and humans are deregulated in a specific cell type single-cell RNA-sequencing (scRNA-seq) data was required. A preprocessed scRNA-seq data set of cirrhotic patients and healthy controls generated by Ramachandran et al. [-@ramachandran_2019] was downloaded from https://datashare.is.ed.ac.uk/handle/10283/3433. The single-cells were annotated by 11 different cell types: MPs, T Cells, ILCs, endothelia, B cells, pDCs, plasma B cells, mast cells, mesenchyme, cholangiocytes, hepatocytes. 
This data set was stored in a Seurat version 2 object and was manually transformed to a Seurat version 3 object, to make it compatible with the latest version of the R package Seurat (version 3.2.3; @butler_2018) Subsequently, a differential gene expression analysis was performed for each cell type individually between cirrhotic and healthy individuals using the function `Seurat::FindAllMarkers()`. A gene was considered differentially expressed with an absolute logFC $\ge$ 0.25 and an FDR $\le$ 0.05.

### Cross-species mapping of gene symbols
To map MGI gene symbols to HGNC symbols or vice versa the R/Bioconductor package biomaRt (version 2.44.0; @durinck_2009) was used, which itself queries the Ensembl Archive Release 99 from January 2020. Gene that did not match were discarded and in case of an ambiguous mapping, the gene with the highest absolute log-fold change was selected.

### Accessing differentially expressed genes from published mouse models of CLD
The differentially expressed genes of 9 published mouse models of CLD were extracted from Supplementary Table 2 of [@teufel_2016]. The 9 mouse models are defined as: WTD = Western-type diet; HF12/18/30 = High fat diet for 12, 18 and 30 weeks; STZ12/18 = Streptozocin diet for 12 and 18 weeks ; MCD4/8 = Methionine- and choline-deficient diet for 4 and 8 weeks; PTEN = Phosphatase and tensin homologue deleted on chromosome 10 knockout mice

Teufel et al. [-@teufel_2016] defined a gene differentially expressed with FDR $\le$ 0.05 and logFC $\ge$ $log_2$(1.5). Hence, this cutoff is also applied for the other mouse models and patients cohorts to make them compared to these chronic mouse models.

### Gene browser for comparison of human and mouse liver disease
The online gene browser was built with the R package shiny (version 1.6.0; https://shiny.rstudio.com). The code to generate the application is freely available on [GitHub](https://github.com/saezlab/liver-disease-atlas-app).

## Supplementary Figures

```{r liver-sfig-1, fig.cap = "Overview of the study cohorts. A. Number of analyzed mice per mouse model (control/treated). B. Number of analyzed patients per cohort (control/disease)."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 1.png"))
```

```{r liver-sfig-2, fig.cap = "Gene expression changes after induction of chronic liver damage in mice by repeated administration of 1 g/kg b.w. CCl~4~ twice weekly for up to 12 months. A. PCA plot contextualized by time points and treatments. B. Volcano plots of months 2, 6 and 12. C. Genes with the highest log-fold changes."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 2.png"))
```

```{r liver-sfig-3, fig.cap = "Expression changes in an acute liver damage mouse model following a single administration of 300 mg/kg b.w. acetaminophen (APAP). A. Volcano plots at hours 1 and 6, and on days 4, 6, 8, 16 after APAP administration. B. Genes with the highest log-fold changes."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 3.png"))
```

```{r liver-sfig-4, fig.cap = "Expression changes in an acute liver damage mouse model induced by administration of a single administration of CCl~4~. A. Experimental design. B. Histological analysis with hematoxylin and eosin (HE) staining, fibrosis grade visualized by Sirius red, and infiltration of immune cells by CD45. The images show induction of pericentral liver lobule damage but without fibrosis development. Scale bars: 100 µm (HE; Sirius red) and 50 µm (CD45). C. Clinical chemistry with alanine transaminase (ALT), aspartate transaminase (AST) activities in plasma. D. PCA analysis of global expression changes. E. Volcano plots at12 hours, and on days 1 and 2 after CCl~4~ administration. F. Genes with the highest log-fold changes. G. Time-resolved clustering of deregulated genes. The panels B and C were provided by Ahmed Ghallab."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 4.png"))
```

```{r liver-sfig-5, fig.cap = "Expression changes in an acute liver damage mouse model following a single administration of CCl~4~. A. Volcano plots at 2 hours and on days 4, 6, 8, 16 after CCl~4~ administration. B. Genes with the highest log-fold changes."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 5.png"))
```

```{r liver-sfig-6, fig.cap = "Characterization and expression changes of mouse liver tissue after two-third hepatectomy. A. Experimental design. B. Transient increase in cell proliferation after hepatectomy based on BrDU staining and lobule area at different time periods after hepatectomy. C. Tissue morphology visualized by co-staining of bile canaliculi (DPPIV; green), pericentral hepatocytes (glutamine synthetase, white) and sinusoidal endothelial cells (yellow). D. PCA analysis of global expression changes. E. Volcano plots at 12 hours, and on days 1 and 2 after partial hepatectomy. F. Genes with the highest log-fold changes. G. Time-resolved clustering of deregulated genes. The panels B and C were provided by Ahmed Ghallab."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 6.png"))
```

```{r liver-sfig-7, fig.cap = "Characterization and expression changes of mouse liver tissue after two-third hepatectomy. A. Volcano plots at hours 1 and 6, and on days 3 and 4, weeks 1 and 2, and months 1 and 3 after partial hepatectomy. B. Genes with the highest log-fold changes."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 7.png"))
```

```{r liver-sfig-8, fig.cap = "Gene expression changes at different time intervals after induction of obstructive cholestasis by bile duct ligation (BDL). A. Experimental design. B. Histological analysis with hematoxylin and eosin (HE) staining, fibrosis grade visualized by Sirius red, and infiltration of immune cells by CD45. The images show bile infarct formation on days 1 and 3, and periportal fibrosis on day 21. Scale bars: 50 µm (HE; CD45) and 200 µm (Sirius red). C. Clinical chemistry with alanine transaminase (ALT), aspartate transaminase (AST) and alkaline phosphatase activities in plasma. D. PCA analysis of global expression changes. E. Volcano plots on days 1, 3 and 7 after BDL. F. Genes with the highest log-fold changes. G. Time-resolved clustering of deregulated genes. The panels B and C were provided by Ahmed Ghallab."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 8.png"))
```

```{r liver-sfig-9, fig.cap = "Expression changes in the chronic stage after induction of obstructive cholestasis by bile duct ligation (BDL). A. Volcano plots on day 21 after BDL. B. Genes with the highest log-fold changes."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 9.png"))
```

```{r liver-sfig-10, fig.cap = "Identification of the time point with the most deregulated expression profile after induction of acute liver injury based on the distance to the respective controls in PCA space along principle component 1 (PC1). Identified time point is colored in green."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 10.png"))
```

```{r liver-sfig-11, fig.cap = "Pool of unified acute and chronic genes demonstrating their respective consistent direction of regulation as indicated by t-statistic."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 11.png"))
```

```{r liver-sfig-12, fig.cap = "Characterization of exclusive acute genes. A. Heatmap of top 100 exclusive acute genes. B-D. Overrepresented transcription factors identified by DoRothEA (B), pathways obtained by PROGENy (C), and GO terms (D) in the upregulated exclusive acute genes. E-F. Same as B-D but for the downregulated exclusive acute genes."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 12.png"))
```

```{r liver-sfig-13, fig.cap = "A-B Overrepresented transcription factors identified by DoRothEA (A), and GO terms (B) in the set of commonly downregulated genes in acute and chronic mouse models."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 13.png"))
```

```{r liver-sfig-14, fig.cap = "Pairwise comparison of the similarity of the 500 deregulated genes per human contrast. Similarity is computed with the Jaccard index."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 14.png"))
```

```{r liver-sfig-15, fig.cap = "A-B Overrepresented pathways obtained by PROGENy (A), and GO terms (B) in the set of consistently downregulated genes between human and mouse."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 15.png"))
```

```{r liver-sfig-16, fig.cap = "Aldo-keto reductase (AKR1B10) and annexin V (ANXA5) increase in chronic liver disease of mice and humans. A. Immunostaining of AKR1B10 and ANXA5 at different stages during chronic liver disease progression induced by chronic CCl~4~ administration in mice; AKR1B10 shows clusters of positive signals at 12 months (arrows); ANXA5 stained positive in the progressive ductular reaction. B. AKR1B10 and ANXA5 immunostaining in fibrotic tissues of human patients showing similar expression patterns as in mice. Scale bars: 100 µm. The entire figure was provided by Ahmed Ghallab."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 16.png"))
```

```{r liver-sfig-17, fig.cap = "A. Number of deregulated genes for the chronic CCl~4~ mouse models complemented by 9 additional publicly available mouse models of chronic liver diseases. B Similarity of the significantly deregulated genes between all chronic mouse models and the human data. Similarity is computed as overlap coefficient."}
knitr::include_graphics(here::here("data/liver_disease_atlas/Supplementary Figure 17.png"))
```


