# Other contributions {#other-contrib}

<!-- required to display altmetric badge -->
<script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>

In this chapter, I will briefly present my contributions to other projects that are not directly related to the topic of this thesis, but still made up an essential part of my Ph.D. At the time of writing, all of the manuscripts listed have either been published in, or at least submitted to, peer-reviewed journals. The R-packages have been accepted at [Bioconductor](https://www.bioconductor.org/), an open source repository for bioinformatics tools.

```{r "save-google-bioc-stats", eval=FALSE, include=FALSE}
# google analytics
# options(gargle_oauth_email = "cholland2408@gmail.com")
# options(gargle_oauth_email = TRUE)
# googledrive::drive_auth(email = TRUE)
# options(gargle_quiet = TRUE)
ga_auth(email = "cholland2408@gmail.com")
ga_property_list = ga_account_list() %>%
  filter(accountName == "Saezlab") %>%
  select(webPropertyName, viewId) %>%
  group_by(webPropertyName) %>%
  summarise(tmp = list(viewId)) %>%
  deframe() 

access_ga = function(id) {
  google_analytics(id,
                   #"219210421",
                   date_range = c("2017-05-01", "today"),
                   metrics = "sessions",
                   dimensions = c("date","country", "city"),
                   max = -1) %>%
    as_tibble() %>%
    mutate(across(where(is.character), na_if, y = "(not set)")) %>%
    rename(region = country) %>%
    # recode  certain entries to match countries in the world data set 
    # hong kong is not available in the world dataset
    mutate(region = recode(str_trim(region),
                           "United States" = "USA",
                           "United Kingdom" = "UK",
                           "Czechia" = "Czech Republic"))
}

# save google analytics data for later use
ga_pages = c("FUNKI App", "CEVIChE","ReHeat","CKD App", "dorothea", "progeny")
ga_stats = ga_property_list[ga_pages] %>% 
  map(access_ga)

# add timestamp
ga_stats$timestamp = my_today()

saveRDS(ga_stats, here("data/contributions/google_analytics_stats.rds"))

# google scholar
id = "h_y2uTwAAAAJ"
publication_list = scholar::get_publications(id) %>%
  as_tibble()

saveRDS(list(p = publication_list, timestamp = my_today()), 
        here("data/contributions/google_scholar_stats.rds"))

# bioconductor
dorothea = read_delim("https://www.bioconductor.org/packages/stats/data-experiment/dorothea/dorothea_stats.tab", 
                        delim = "\t")

progeny = read_delim("http://bioconductor.org/packages/stats/bioc/progeny/progeny_stats.tab", 
                     delim = "\t") 

saveRDS(list(dorothea = dorothea, progeny=progeny, timestamp = my_today()),
        here("data/contributions/bioconductor_stats.rds"))
```

```{r load-google-bioc-stats}
# google analytics
ga_stats = readRDS(here("data/contributions/google_analytics_stats.rds"))

funki = ga_stats$`FUNKI App`
ceviche = ga_stats$CEVIChE
reheat = ga_stats$ReHeat
ckd = ga_stats$`CKD App`
dorothea = ga_stats$dorothea
progeny = ga_stats$progeny


plot_worldmap = function(property) {
  world = map_data("world")
  property %>%
    group_by(region) %>%
    summarise(n = sum(sessions)) %>%
    mutate(prop = n/sum(n)) %>%
    full_join(world, by="region") %>%
    filter(region != "Antarctica") %>%
    ggplot(x, mapping = aes(x=long, y=lat, group = group)) +
    geom_polygon(aes(fill = n)) +
    scale_fill_distiller(palette ="RdBu", direction = -1) +
    theme_void() +
    # coord_fixed(1.3) +
    # theme(legend.position = "bottom") +
    labs(fill = str_wrap("Number of visitors per country", width = 12))
}

plot_timecourse = function(property) {
  property %>%
    group_by(week = as_date(cut(date, "week")),
             year = year(date)) %>%
    summarise(n = sum(sessions)) %>%
    ggplot(aes(x=week, y=n)) +
    geom_line() +
    scale_x_date() +
    ylim(c(0, NA)) +
    labs(x = "Time", y="Number of weekly visitors")
}

plot_cumsum = function(property) {
  property %>%
    group_by(week = as_date(cut(date, "week"))) %>%
    summarise(n = sum(sessions)) %>%
    mutate(csum = cumsum(n)) %>%
    ggplot(aes(x=week, y=csum)) +
    geom_line() +
    scale_x_date() +
    ylim(c(0, NA)) +
    labs(x = "Time", y="Total number of visitors")
}

weekly_visitors = function(property) {
  property %>%
    group_by(week = as_date(cut(date, "week")),
             year = year(date)) %>%
    summarise(n = sum(sessions)) %>%
    ungroup() %>%
    summarise(m = ceiling(median(n))) %>%
    pull(m)
}

total_visitors = function(property) {
  property %>%
    summarise(t = sum(sessions)) %>%
    pull(t)
}

start_date = function(property) {
  property %>%
    slice_min(date, with_ties = FALSE) %>%
    pull(date) %>%
    format('%d %B %Y')
}

num_countries = function(property) {
  property %>%
    summarize(n = n_distinct(region)) %>%
    pull() %>%
    format(big.mark=",")
}

# google scholar
gs = readRDS(here("data/contributions/google_scholar_stats.rds"))

get_citation_num = function(id) {
  gs$p %>%
    filter(stringr::str_detect(title, pattern = id)) %>%
    pull(cites) %>%
    format(big.mark=",")
}

# bioconductor
bc = readRDS(here("data/contributions/bioconductor_stats.rds"))

num_downloads = function(package) {
  total = package %>%
    filter(Month != "all") %>%
    summarise(s = sum(Nb_of_downloads)) %>%
    pull(s) %>%
    format(big.mark=",")
  
  # if the year 2017 appears in the dataset it must be the progeny package
  # for the progeny package compute the total number of downloads and the number
  # of downloads since I started to contribute (May 2020)
  if (2017 %in% package$Year) {
    release_3_11 = package %>%
      filter(Month != "all") %>%
      mutate(date = dmy(str_c("1", Month, Year, sep = " "))) %>%
      mutate(month = month(date, label = T),
             year = year(date)) %>%
      filter(date >= as_date("2020-05-01")) %>%
      summarise(s = sum(Nb_of_downloads)) %>%
      pull(s) %>%
      format(big.mark=",")
    
    return(c(total, release_3_11))
  } else {
    return(total)
  }
}
```


## Manuscripts
### Benchmark and integration of resources for the estimation of human transcription factor activity {.unlisted .unnumbered}
<div data-badge-popover="right" data-badge-type="donut" data-doi="10.1101/gr.240663.118" data-hide-no-mentions="true" class="altmetric-embed"></div>

This work was led by Luz Garcia-Alonso, a former postdoc of Julio Saez-Rodriguez group, and was [published in Genome Research in 2019 ](http://www.genome.org/cgi/doi/10.1101/gr.240663.118). As of `r gs$timestamp` this work gathered `r get_citation_num("Benchmark and")` citations on [Google Scholar](https://scholar.google.com/scholar?hl=de&as_sdt=0%2C5&q=Benchmark+and+integration+of+resources+for+the+estimation+of+human+transcription+factor+activities&btnG=).

Transcriptional regulation is a key mechanism that allows cells to react adequately to external and internal stimuli or perturbation. Guided through transcription factors (TFs) the transcription and thus expression of genes can be activated or repressed that finally enables cells to adapt to the changing environment. The identification of regulatory interactions between TFs and genes is a highly active research area. Hence, there exist various resources that provide a mapping between a TF and its transcriptional targets (i.e. regulon). These regulons can be derived from various techniques spanning from web lab experiments such as ChiP-seq to in silico predictions via network reconstruction algorithms.
In this work, four different lines of evidence of regulon construction were integrated into a consensus TF-target network, that was finally benchmarked via TF activity predictions. 

My contribution focussed on the inference of TF-target interactions from gene expression data with the tool ARACNe. As a resource for the gene expression data, I utilized the GTEx portal and the TCGA program. For each tissue/cancer type, I inferred an individual gene regulatory network. Subsequently, I performed a systematic comparison among the tissue and cancer-specific networks with respect to regulon sizes and target similarities. Finally, those tissue/cancer-specific contributed to the consensus gene regulatory network. Furthermore I developed a web application with [R-Shiny](https://shiny.rstudio.com/) to infer TF activities from user provided gene expression data and to visualize the results. The app is named [FUNKI](https://saezlab.shinyapps.io/funkiapp/) and is visited on average  `r weekly_visitors(funki)` times per week (Fig. \@ref(fig:funki-stats)A). Altogether  `r total_visitors(funki)` visitors from `r num_countries(funki)` different countries have visited the app since  `r start_date(funki)` (Fig. \@ref(fig:funki-stats)B and C). 

```{r funki-stats, fig.asp=0.9, fig.cap = "Visitor statistics for [_FUNKI_](https://saezlab.shinyapps.io/funkiapp/). **A** Number of visitors per week. **B** Total number of visitors. **C** Origin of the visitors.", fig.scap = "Visitor statistics for _FUNKI_."}
ts = plot_timecourse(funki) 
m = plot_worldmap(funki)
cs = plot_cumsum(funki)

((ts + cs) / m) + plot_annotation(tag_levels = "A") +
  labs(caption = str_c("as of ", ga_stats$timestamp))
```

### Signatures of cell death and proliferation in perturbation transcriptomics data - from confounding factor to effective predictions {.unlisted .unnumbered}
<div data-badge-popover="right" data-badge-type="donut" data-doi="10.1093/nar/gkz805" data-hide-no-mentions="true" class="altmetric-embed"></div>

This work was led by Bence Szalai, a former postdoc of Julio Saez-Rodriguez group, and was [published in Nucleic Acid Research in 2019](https://doi.org/10.1093/nar/gkz805). As of `r gs$timestamp` this work gathered `r get_citation_num("cell death")` citations on [Google Scholar](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=Signatures+of+cell+death+and+proliferation+in+perturbation+transcriptomics+data+-+from+confounding+factor+to+effective+predictions&btnG=).

Drug or gene perturbation experiments are a valuable approach to identify perturbation specific signatures on the transcriptome level, which are typically further characterized by functional genomics analyses. The recent LINCS-L1000 panel generated more than 1,000,000 perturbation signatures, derived from more than 20,000 different perturbagenes. This work claims that perturbation signatures are a combination of two signatures types. First, the signature related to the desired perturbation effect and second a cell viability signature (cell death or proliferation) which is unrelated to the actual perturbation. By
integrating perturbation signatures with large cell viability studies it was possible to identify a specific and consistent cell viability signature. Among other areas of application that are reported in this publication, this signature is used to predict cell viability from gene expression data. 

I contributed to this project by developing a user-friendly  [R-Shiny](https://shiny.rstudio.com/) web application that enables users to predict cell viability from their custom gene expression data. Next to the prediction task, this application can also be used to browse other static data from this publication, e.g. the measured cell viability from different resources that matches perturbation data. The app is named [CEViChE](https://saezlab.shinyapps.io/ceviche/) (CEll VIability Calculator from gene Expression) and is visited on average  `r weekly_visitors(ceviche)` times per week (Fig. \@ref(fig:ceviche-stats)A). Altogether  `r total_visitors(ceviche)` visitors from `r num_countries(ceviche)` different countries have visited the app since  `r start_date(ceviche)` (Fig. \@ref(fig:ceviche-stats)B and C). 

```{r ceviche-stats, fig.asp=0.9, fig.cap = "Visitor statistics for [_CEVIChE_](https://saezlab.shinyapps.io/ceviche/). **A** Number of visitors per week. **B** Total number of visitors. **C** Origin of the visitors.", fig.scap = "Visitor statistics for _CEVIChE_."}
ts = plot_timecourse(ceviche) 
m = plot_worldmap(ceviche)
cs = plot_cumsum(ceviche)

((ts + cs) / m) + plot_annotation(tag_levels = "A") +
  labs(caption = str_c("as of ", ga_stats$timestamp))
```


### Influence of Liver Fibrosis on Lobular Zonation {.unlisted .unnumbered}
<div data-badge-popover="right" data-badge-type="donut" data-doi="10.3390/cells8121556" data-hide-no-mentions="true" class="altmetric-embed"></div>

This work was led by Ahmed Ghallab, a collaborator from the LiSyM consortium, and was [published in Cells in 2019](https://dx.doi.org/10.3390/cells8121556). As of `r gs$timestamp` this work gathered `r get_citation_num("Lobular Zonation")` citations on [Google Scholar](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=Influence+of+Liver+Fibrosis+on+Lobular+Zonation&btnG=).

Liver architecture is composed of so-called liver lobules, that serve as the main building blocks of the liver tissue. A liver lobule itself is mainly composed of hepatocytes building the parenchyma and are arranged in a hexagonal shape. In a healthy liver, a liver lobule is spatially separated along the porto-central axis into the pericentral region near the central vein and the periportal region near the portal vein. While this zonation has been thoroughly investigated in the healthy liver, it is still an open question of how liver fibrosis influences the zonation. This project aims to shed light on this question from various perspectives. 

My contribution focussed on the analysis of time-resolved gene expression data of a mouse model developing chronic liver damage through regular CCl4 treatment. The transcriptomic profile was measured after 2, 6, and 12 months after the initial injection. First I described the transcriptomic landscape of this mouse model performing differential gene expression analysis and downstream pathway and TF analyses. Subsequently, I set out to investigate the putative change in the expression of marker genes of the pericentral and periportal regions. Hence, I identified these gene markers by integrating publicly available bulk and single-cell transcriptomics data, each providing different sets of periportal and pericentral genes. Enrichment analyses with those consensus periportal and pericentral gene sets and the mouse model revealed that pericentral genes were consistently downregulated across all time points, while periportal genes were consistently up-regulated. This indicates that during liver fibrosis the pericentral region adopts periportal characteristics, which was referred to as periportalization of the liver. 

Considering these results and further validation experiments performed from others showed that liver fibrosis alters the lobular zonation strongly.


### A Consensus Transcriptional Landscape of Human End-Stage Heart Failure {.unlisted .unnumbered}
<div data-badge-popover="right" data-badge-type="donut" data-doi="10.1101/2020.05.23.20110858" data-hide-no-mentions="true" class="altmetric-embed"></div>

This work was co-led by Ricardo Ramirez Flores and Jan Lanzer, both Ph.D. students in Julio Saez-Rodriguez group, and is [posted to medRxiv](https://doi.org/10.1101/2020.05.23.20110858) and submitted to Circulation at the time of writing. As of `r gs$timestamp` this work gathered `r get_citation_num("Heart Failure")` citations on [Google Scholar](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=A+Consensus+Transcriptional+Landscape+of+Human+End-Stage+Heart+Failure&btnG=).

Heart failure is a severe health issue with a global prevalence of more than 37 million individuals. Worldwide, scientists prospect for novel treatment and diagnostic options. One large branch of those efforts is to study the gene regulation in heart failure. Over the years many large transcriptomic studies of patients suffering from heart failure have been performed. However, the agreement of marker genes for heart failure among studies is poor. Hence, there is a need for a comprehensive and systematic meta-analysis integrating independent clinical cohorts. 
This project collated 16 different publicly available transcriptomic heart failure data sets with the aim to define a consensus gene signature of heart failure. While the top differentially expressed genes of individual studies overlapped poorly, it was possible to show that the general trend of gene regulation was conserved across studies. Finally, the authors retrieved a consensus and weighted gene signature of heart failure, which was analyzed and characterized thoroughly. As the genes are weighted in the signature, a possible application is to investigate whether a custom gene set is significantly enriched in the heart failure signature. 

My contribution to this project was to develop a user-friendly web application with [R-Shiny](https://shiny.rstudio.com/)  that provides an intuitive interface for this enrichment task. In addition, the web page displays static results of the characterization of the heart failure signature and metadata of the analyzed studies. The app is named [ReHeat](https://saezlab.shinyapps.io/reheat/) (REFerence for the HEArt failure Transcriptome) and is visited on average  `r weekly_visitors(reheat)` times per week (Fig. \@ref(fig:reheat-stats)A). Altogether  `r total_visitors(reheat)` visitors from `r num_countries(reheat)` different countries have visited the app since  `r start_date(reheat)` (Fig. \@ref(fig:reheat-stats)B and C).


```{r "reheat-stats", fig.asp=0.9, fig.cap = "Visitor statistics for [_ReHeat_](https://saezlab.shinyapps.io/reheat/). **A** Number of visitors per week. **B** Total number of visitors. **C** Origin of the visitors.", fig.scap = "Visitor statistics for _ReHeat_."}
ts = plot_timecourse(reheat) 
m = plot_worldmap(reheat)
cs = plot_cumsum(reheat)

((ts + cs) / m) + plot_annotation(tag_levels = "A") +
  labs(caption = str_c("as of ", ga_stats$timestamp))
```

### A functional landscape of CKD entities from public transcriptomic data {.unlisted .unnumbered}
<div data-badge-popover="right" data-badge-type="donut" data-doi="10.1016/j.ekir.2019.11.005" data-hide-no-mentions="true" class="altmetric-embed"></div>

This work was led by Ferenc Tajti, a former intern of Julio Saez-Rodriguez group, and was [published in Kidney International Reports in 2020](https://doi.org/10.1016/j.ekir.2019.11.005). As of `r gs$timestamp` this work gathered `r get_citation_num("CKD")` citations on [Google Scholar](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=A+functional+landscape+of+CKD+entities+from+public+transcriptomic+data&btnG=).

Chronic kidney disease (CKD) can arise from many different origins, such as diabetic nephropathy or hypertensive nephropathy. Regardless of the underlying etiology the repetitive injury leads to fibrosis and finally kidney failure. This study sets out to investigate and to characterize the changes in gene expression for the different CKD entities. Therefore, a pipeline was developed to describe the transcriptomic landscape of publicly available CKD studies. This pipeline comprises data integration, gene expression analysis, and functional characterization via pathway and transcription factor analysis. Potential biomarkers such as the transcription factor FOXM1 were experimentally validated. Furthermore, the individual gene signatures from the CKD entities were used for drug repositioning, which revealed the FDA approved drug nilotinib as a potential treatment for CKD.

I contributed to this project by developing an [R-Shiny](https://shiny.rstudio.com/) web application that visualizes the key findings of the project. Besides, interested users can interactively explore the data and test their hypotheses, such as investigating the role of a specific gene or pathway of interest in the CKD studies. The app is named [CKD-App](https://saezlab.shinyapps.io/ckd_landscape/) (Chronic Kidney Disease-App) and is visited on average  `r weekly_visitors(ckd)` times per week (Fig. \@ref(fig:ckd-stats)A). Altogether  `r total_visitors(ckd)` visitors from `r num_countries(ckd)` different countries have visited the app since  `r start_date(ckd)` (Fig. \@ref(fig:ckd-stats)B and C).




```{r "ckd-stats", fig.asp=0.9, fig.cap = "Visitor statistics for [_CKD App_](https://saezlab.shinyapps.io/ckd_landscape/). **A** Number of visitors per week. **B** Total number of visitors. **C** Origin of the visitors.", fig.scap = "Visitor statistics for _CKD App_."}
ts = plot_timecourse(ckd) 
m = plot_worldmap(ckd)
cs = plot_cumsum(ckd)

((ts + cs) / m) + plot_annotation(tag_levels = "A") +
  labs(caption = str_c("as of ", ga_stats$timestamp))
```

### NRF2/KEAP1 in hepatocytes controls fibro- and carcinogenesis in chronic liver disease {.unlisted .unnumbered}

This work was led by Antje Mohs, a collaborator from University Hospital RWTH Aachen, and is under review in the Journal of Hepatology at the time of writing.

Enduring inflammation of the liver leads to fibrosis and thus to the development of chronic liver diseases. This inflammation process induces also oxidative stress which might a potential driver of liver fibrosis progression. Typically cells protect themselves from oxidative stress via the Nuclear factor erythroid 2-related factor 2 (NRF2)/Kelch-like ECH-associated protein 1 (KEAP1) system. Under oxidative stress conditions, the transcription factor NRF2 translocates into the nucleus and activates the expression of its transcriptional targets, which are involved in oxidative induced stress response. To investigate the role of NRF2 for the progression of chronic liver diseases, patient cohorts, and appropriate mouse models were analyzed.

I contributed to this project by analyzing the expression and activity of NRF2 in a pediatric and an adult nonalcohol fatty liver disease (NAFLD) cohort. In both cohorts, I was able to show that the activity of NRF2 is significantly and positively associated with the grade of inflammation of the liver. However, the grade of steatosis had no significant impact on NRF2 activity. These results and further functional analyses of NRF2 in mouse models (performed by other co-authors) revealed that “NRF2 activation in chronic liver disease is protective by ameliorating fibrogenesis, initiation, and progression of hepatocellular carcinogenesis.”


### Imbalanced gut microbiota fuels HCC development by shaping the hepatic inflammatory microenvironment {.unlisted .unnumbered}

This work was co-led by Kai Markus Schneider and Antje Mohs, both collaborators from University Hospital RWTH Aachen, and is submitted to Cancer Cell at the time of writing.

Via the blood supply through the portal vein, many pathogens, and microbiota associated molecular patterns are washed from the gut into the liver. Those patterns activate anti-inflammatory pathways such as NF-kB which leads to the progression of liver disease. Hence, gut microbiota may also shape the development of hepatocellular carcinoma which is a severe and widespread health issue, with still limited therapeutic options.

I contributed to this hypothesis by analyzing a clinical cohort with patients who suffered from advanced liver cirrhosis that underwent liver transplantation as well as healthy controls. For each patient, a paired transcriptomic and metagenomic data were profiled. First I characterized the transcriptomic landscape of this patient cohort by differential gene expression analysis and downstream pathway and transcription factor analyses. Several fibro-inflammatory pathways and oncogenic TFs were found to be upregulated in cirrhosis patients. In order to integrate the transcriptomic and metagenomic data, I correlated genes, TF, pathways with 16s rDNA abundance derived from the metagenomic profiles which can be interpreted as a proxy for bacterial translocation. This analysis revealed that well-known HCC related pathways, TF, and genes correlated significantly with bacterial translation. 
Finally, I dissected the hepatic cellular landscape via a gene expression-based cell type enrichment method, showing an overall higher abundance of immune and stroma cells in cirrhosis patients. 

Considering these results and further analyses of various mouse models performed by other co-authors revealed “that gut microbiota closely shape the hepatic inflammatory microenvironment defining new approaches for cancer prevention and therapy.”


## R-packages

### dorothea {.unlisted .unnumbered}
*dorothea* is an R-package developed by me which is based on the publications [Benchmark and integration of resources for the estimation of human transcription factor activity](http://www.genome.org/cgi/doi/10.1101/gr.240663.118) and [Transfer of regulatory knowledge from human to mouse for functional genomics analysis](https://doi.org/10.1016/j.bbagrm.2019.194431). The package contains manually curated human and mouse transcription factor regulons, that can be coupled with any statistical method that aims to analyse gene sets to infer transcription factor activity from gene expression data. Preferably *dorothea*’s regulons are coupled with the statistical method *viper*. Therefore, I developed a convenient wrapper that combines *viper* with *dorothea*’s regulons. *dorothea* was accepted as an “Experiment Package” with the release of Bioconductor 3.11 on 28 April 2020.

So far the *dorothea* package has been downloaded  `r num_downloads(bc$dorothea)` times (Fig. \@ref(fig:dorothea-bc-stats)). 

```{r dorothea-bc-stats, fig.cap = "Download statistics of the [_dorothea_ Bioconductor package](https://bioconductor.org/packages/release/data/experiment/html/dorothea.html).", fig.scap = "Download statistics of the _dorothea_ Bioconductor package"}
bc$dorothea %>%
  filter(Month != "all") %>%
  mutate(date = dmy(str_c("1", Month, Year, sep = " "))) %>%
  mutate(month = month(date, label = T),
         year = year(date)) %>%
  ggplot(aes(x=month, y=Nb_of_downloads)) +
  geom_col() +
  facet_wrap(~year) +
  labs(x = "Month of the year", y = "Number of monthly downloads",
       caption = str_c("as of ", bc$timestamp))
```

The corresponding [GitHub page](https://saezlab.github.io/dorothea/) is visited on average  `r weekly_visitors(dorothea)` times per week (Fig. \@ref(fig:dorothea-gh-stats)A). Altogether  `r total_visitors(dorothea)` visitors from `r num_countries(dorothea)` different countries have visited the GitHub page since  `r start_date(dorothea)` (Fig. \@ref(fig:dorothea-gh-stats)B and C).

```{r dorothea-gh-stats, fig.asp=0.9, fig.cap = "Visitor statistics for [_dorothea_'s GitHub page](https://saezlab.github.io/dorothea/). **A** Number of visitors per week. **B** Total number of visitors. **C** Origin of the visitors.", fig.scap = "Visitor statistics for _dorothea_'s GitHub page."}
ts = plot_timecourse(dorothea) 
m = plot_worldmap(dorothea)
cs = plot_cumsum(dorothea)

((ts + cs) / m) + plot_annotation(tag_levels = "A") + 
  labs(caption = str_c("as of ", ga_stats$timestamp))
```

### progeny {.unlisted .unnumbered}

*progeny* is an R package that was originally developed by Michael Schubert, a former Ph.D. student of Julio Saez-Rodriguez group as part of the work [Perturbation-response genes reveal signaling footprints in cancer gene expression](https://doi.org/10.1038/s41467-017-02391-6). It was accepted as a “Software package” with the release of Bioconductor 3.6 on 31 October 2017. Its objective is to infer pathway activities from human gene expression data. 
Together with Bence Szalai, a former postdoc of Julio Saez-Rodriguez group, I contributed to this project by expanding the collection of pathways and adapting the tool to mouse gene expression data as part of the work [Transfer of regulatory knowledge from human to mouse for functional genomics analysis](https://doi.org/10.1016/j.bbagrm.2019.194431). These new features were integrated into the *progeny* package with the release of Bioconductor 3.11 on 28 April 2020 which was led by Alberto Valdeolivas, a postdoc of Julio Saez-Rodriguez group. 

Since the release of Bioconductor 3.11 the *progeny* package has been downloaded  `r num_downloads(bc$progeny)[2]` times (since Biocondutor release 3.6 `r num_downloads(bc$progeny)[1]` times; Fig. \@ref(fig:progeny-bc-stats)).

```{r "progeny-bc-stats", fig.cap = "Download statistics of the [_progeny_ Bioconductor package](https://bioconductor.org/packages/release/bioc/html/progeny.html). Red colored bars indicate the time from which I started to contributed to the development.", fig.scap = "Download statistics of the _progeny_ Bioconductor package"}
bc$progeny %>%
  filter(Month != "all") %>%
  mutate(date = dmy(str_c("1", Month, Year, sep = " "))) %>%
  mutate(month = month(date, label = T),
         year = year(date)) %>%
  mutate(c = case_when(date >= as_date("2020-05-01") ~ "yes",
                       TRUE ~ "no")) %>%
  ggplot(aes(x=month, y=Nb_of_downloads, fill=c)) +
  geom_col() +
  facet_wrap(~year) +
  scale_fill_manual(values = aachen_color(c("black75","bordeaux"))) +
  labs(x = "Month of the year", y = "Number of monthly downloads",
       caption = str_c("as of ", bc$timestamp)) +
  theme(legend.position = "none")
```

The corresponding [GitHub page](https://saezlab.github.io/progeny/) is visited on average  `r weekly_visitors(progeny)` times per week (Fig. \@ref(fig:progeny-gh-stats)A). Altogether  `r total_visitors(progeny)` visitors from `r num_countries(progeny)` different countries have visited the GitHub page since  `r start_date(progeny)` (Fig. \@ref(fig:progeny-gh-stats)B and C).


```{r "progeny-gh-stats", fig.asp=0.9, fig.cap = "Visitor statistics for [_progeny_'s GitHub page](https://saezlab.github.io/progeny/). **A** Number of visitors per week. **B** Total number of visitors. **C** Origin of the visitors.", fig.scap = "Visitor statistics for _progeny_'s GitHub page."}
ts = plot_timecourse(progeny) 
m = plot_worldmap(progeny)
cs = plot_cumsum(progeny)

((ts + cs) / m) + plot_annotation(tag_levels = "A") +
  labs(caption = str_c("as of ", ga_stats$timestamp))
```

