Transcriptomics is one of the most used omics technology that studies the abundance of RNA transcripts. High-throughput techniques such as microarrays and RNA-sequencing enable the relatively easy and inexpensive collection of bulk gene expression profiles from any biological conditions. Recently, also the transcriptome of single cells can be efficiently captured via novel single-cell RNA-sequencing technologies. Functional analysis of bulk or single-cell gene expression data has been proven to be a powerful approach as they summarize the large and noisy gene expression space into a smaller number of biologically meaningful features such as pathway and transcription factor activities. In the first part of this thesis, I expanded the scope on the pathway analysis tool PROGENy and the transcription factor analysis tool DoRothEA through thorough benchmarking pipelines. First I transferred their regulatory knowledge from human to mouse to enable the functional characterization of gene expression profiles from the model organism Mus Musculus. Moreover, I demonstrated the robustness and applicability of both tools on human single-cell RNA-sequencing data. In the second part of this thesis, I focussed on the analysis of gene expression profiles from mice and humans in the context of acute and chronic liver damage. I aimed to identify differences between the acute and chronic scenarios and to access the translational relevance of mouse models. Finally, I identified exclusively and commonly regulated genes of chronic and acute liver damage in mice and a set of genes that were consistently altered in a novel chronic mouse model and patients of chronic liver disease. Application of DoRothEA and PROGENy revealed for those gene sets potential common master regulators and affected signaling pathways.
